test/test_geod.py

```diff
@@ -443,6 +443,17 @@ def test_geodesic_npts(include_initial, include_terminus):
     assert_almost_equal(lonlats, expected_lonlats, decimal=3)
 
 
+def test_geodesic_npts__nan():
```

Comment:
I am wondering whether it wouldn't be nice to safeguard against all nan positions? You could use parametrize for that.

One way would be to do the following:

```
@pytest.mark.parametrize(lon_one, [1.0, np.nan])
@pytest.mark.parametrize(lat_one, [2.0, np.nan])
@pytest.mark.parametrize(lon_two, [3.0, np.nan])
@pytest.mark.parametrize(lat_two, [4.0, np.nan])
def test_geodesic_npts__nan(lon_one, lat_one, lon_two, lat_two):
    coords_list = [lon_one, lat_one, lon_two, lat_two]
    if np.isfinite(coords_list).all():
        return

    geod = Geod(ellps="WGS84")
    assert_array_equal(
        geod.npts(*coords_list, npts=1), [(math.nan, math.nan)]
    )
```

Alternatively, to omit the return statement:
```
from itertools import combinations

@pytest.mark.parametrize(nan_positions, list(combinations([0, 1, 2, 3], 1)) + list(combinations([0, 1, 2, 3], 2)) + list(combinations([0, 1, 2, 3], 3)) + list(combinations([0, 1, 2, 3], 4)))
def test_geodesic_npts__nan(nan_positions):
    geod = Geod(ellps="WGS84")
    coords_list = [1.0, 2.0, 3.0, 4.0]
    for position in nan_positions:
        coords_list[position] = math.nan

    assert_array_equal(
        geod.npts(*coords_list, npts=1), [(math.nan, math.nan)]
    )
```

Reply:
That sounds like a good idea.

Reply:
I went with a simpler approach that I think covers it.